### ISSUE TITLE
Comments Feature in CodiMD

### ISSUE THREAD

**Issue by jakubgs** [hackmdio/codimd#1223]
I see https://hackmd.io has comments for documents, which is pretty cool. Is that feature ever coming to CodiMD?

**Comment by zeigerpuppy** [comment]
It would be fantastic to have a way of supporting a commenting side bar with the following features:

- comments stored in document using [CriticMarkup](https://github.com/CriticMarkup/CriticMarkup-toolkit/blob/master/README.md)
- tools for add/remove/change/highlight which insert syntax in selected passage
- marginal marks in edit and rendered view to show an edit
- buttons for accept/reject of edits
- tracking of a discussion/chat for each edit
   - this is a bit more difficult.  Ideally it would be good to retain data in the CriticMarkup format but it woukd be necessary to add autor tag to comments automatically and also to have the option of collapsing comments in the edit view, or else it could become distracting.

In summary, these features would allow multi-user commenting and editing workflow while retaining consistency with the markdown document and multi-user collaboration.

**Comment by ei8fdb** [comment]
I would be *very* happy if commenting support was added with 1) markdown comment, 2) email address/name input field, as a beginning?

Is there any progress, or indication if this functionality can be added?

**Comment by raphaeljolivet** [comment]
For comment threads, this might bloat the markdown content. 
CriticMarkup should be extended to support reference to external comments, via IDs, as footnotes or separate files (json / separate db)
something like : 
{## 1 ##} ... {## ef18de ##}

**Comment by stgolem** [comment]
I think we should not mix this and CriticMarkup feature. For us it would be great to have discussion on document outside of document, if we can download it as plain file later.

**Comment by yanickxia** [comment]
any update for this feature?

**Comment by Iaskebba** [comment]
I am also curious on an update for this feature? If someone is more familiar with how this feature should be architected I am happy to own it.
